### ISSUE TITLE
pending builds are not kicking

### ISSUE THREAD

**Issue by tardyp** [buildbot/buildbot#2937]
This ticket is a migrated Trac ticket [3265](http://trac.buildbot.net/ticket/3265)

People contributed to the original ticket: @djmitche, `benoit.allard@...`
Ticket created on: `May 26 2015`
Ticket last modified on: `Sep 06 2015`

---

I configured some new builders, and reloaded the configs, builders appear everything looks good. Later I wondered why no build where scheduled on those builders, and why the requests are queueing up. My one slave is up and processing builds from other builders.

Upon a master restart, all pending builds are being kicked.

I have no idea what the problem was / is, and realize that my problem description might not help much.

---

_Comment from_: @dustin
_Date_: `Sep 06 2015`

We had another issue like this around latent slaves with wait_time=0.  Is that the case here?

**Comment by PunChaFeng** [comment]
@seankelly, I encountered the same issue. I can give you some hint.

1. I add only one Latent Slave/Worker and set its `build_wait_timeout` to 0. So, for each job, buildbot should substantiate a new VM and insubstantiate it after build is finished.
1. I queue two jobs at once.
1. The slave picks up the first job and starts substantiate an VM to build.
1. Once the build is finished, buildbot is insubstantiating it and in the meanwhile, `maybeStartBuild()` function is called to ask the slave whether it is ready to pick another job. Since the latent slave is insubstantiating, so the answer is no. The second jobs are not picked and keep `Pending`.

**Comment by tardyp** [comment]
@PunChaFeng which version of buildbot are you working on?

**Comment by PunChaFeng** [comment]
@tardyp  I'm using 0.8.7p1

**Comment by tardyp** [comment]
Buildbot 0.8.7p1 was released on November 21, 2012. A lot happen since then and this bug was fixed.
I you want to use LatentWorker, please use buildbot >0.9.3
